ssapy/compute.py

```diff
@@ -1336,3 +1348,207 @@ def M_v(r_sat, r_earth, r_sun, r_moon=False, radius=0.4, albedo=0.20, sun_Mag=4.
+
+        ax = fig.add_subplot(1, 4, 2)
+        ax.scatter(r_earth[0], r_earth[1], c='Blue', s=10)
+        scatter = ax.scatter(r_sat[:, 0] / RGEO, r_sat[:, 1] / RGEO, c=(2.5 * np.log10((r_sun_sat / (10 * pc_to_m))**2) + sun_Mag) - 2.5 * np.log10(frac_flux_sun), cmap=color_map)
+        colorbar = plt.colorbar(scatter)
+        ax.scatter(r_sun[:, 0] / sun_scale, r_sun[:, 1] / sun_scale, c=plt.cm.Oranges(np.linspace(0.25, 0.75, len(r_sat[:, 0]))), s=10)
+        ax.set_title('Solar M_v')
+        ax.axis('equal')
+
+        ax = fig.add_subplot(1, 4, 3)
+        ax.scatter(r_earth[:, 0], r_earth[:, 1], c='Blue', s=10)
+        scatter = ax.scatter(r_sat[:, 0] / RGEO, r_sat[:, 1] / RGEO, c=(2.5 * np.log10((r_sun_sat / (10 * pc_to_m))**2) + sun_Mag) - 2.5 * np.log10(frac_flux_earth), cmap=color_map)
+        colorbar = plt.colorbar(scatter)
+        ax.scatter(r_sun[:, 0] / sun_scale, r_sun[:, 1] / sun_scale, c=plt.cm.Oranges(np.linspace(0.25, 0.75, len(r_sat[:, 0]))), s=10)
+
+        ax.set_title('Earth M_v')
+        ax.axis('equal')
+
+        ax = fig.add_subplot(1, 4, 4)
+        ax.scatter(r_earth[:, 0], r_earth[:, 1], c='Blue', s=10)
+        scatter = ax.scatter(r_sat[:, 0] / RGEO, r_sat[:, 1] / RGEO, c=(2.5 * np.log10((r_sun_sat / (10 * pc_to_m))**2) + sun_Mag) - 2.5 * np.log10(frac_flux_moon), cmap=color_map)
+        ax.scatter(r_moon[:, 0] / RGEO, r_moon[:, 1] / RGEO, c=plt.cm.Greys(np.linspace(0.5, 1, len(r_sat[:, 0]))), s=5)
+
+        colorbar = plt.colorbar(scatter)
+        ax.set_title('Lunar M_v')
+        ax.axis('equal')
+        plt.show()
+
+    return Mag_v
+
+
+def calc_gamma(r, t):
+    """
+    Calculate the gamma angle between position and velocity vectors in the ITRF frame.
+
+    Parameters:
+    r (numpy.ndarray): The position vectors in the GCRF frame, shaped (n, 3), where n is the number of vectors.
+    t (numpy.ndarray or astropy.time.Time): The times corresponding to the position vectors. Can be an array of GPS seconds or an Astropy Time object.
+
+    Returns:
+    numpy.ndarray: An array of gamma angles in degrees between the position and velocity vectors for each time point.
+
+    Notes:
+    - This function first converts the given position vectors from the GCRF frame to the ITRF frame.
+    - It then calculates the angle between the position and velocity vectors at each time point in the ITRF frame.
+    - If the input time array is an Astropy Time object, it converts it to GPS time before processing.
+    """
+    r_itrf, v_itrf = gcrf_to_itrf(r, t, v=True)
+    if isinstance(t[0], Time):
+        t = t.gps
+    gamma = np.degrees(np.apply_along_axis(lambda x: angle_between_vectors(x[:3], x[3:]), 1, np.concatenate((r_itrf, v_itrf), axis=1))) - 90
+    return gamma
+
+
+
+def moon_normal_vector(t):
+    r = get_body("moon").position(t).T
+    r_random = get_body("moon").position(t.gps + 604800).T
+    return np.cross(r, r_random) / np.linalg.norm(r, axis=-1)
+
+
+def lunar_lagrange_points(t):
```

Comment:
Please add a docstring to this function.

Reply:
If the function below is `lunar_lagrange_points_circular`, what is this function calculating? Please change the name to specify what makes this function unique.